src/libraries/System.Private.CoreLib/src/System/IO/WindowsFileStreamStrategy.cs

```diff
@@ -0,0 +1,307 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using System.Runtime.CompilerServices;
+
+namespace System.IO
+{
+    // this type serves some basic functionality that is common for Async and Sync Windows File Stream Strategies
+    internal abstract class WindowsFileStreamStrategy : FileStreamStrategy
+    {
+        // Error codes (not HRESULTS), from winerror.h
+        internal const int ERROR_BROKEN_PIPE = 109;
+        internal const int ERROR_NO_DATA = 232;
+        protected const int ERROR_HANDLE_EOF = 38;
+        protected const int ERROR_INVALID_PARAMETER = 87;
+        protected const int ERROR_IO_PENDING = 997;
+
+        protected readonly SafeFileHandle _fileHandle; // only ever null if ctor throws
+
+        /// <summary>Whether the file is opened for reading, writing, or both.</summary>
```

Comment:
super nit: Let's consider using triple slash comments only on public APIs. For the rest, double slash.
Non blocking, we can address this later.

Reply:
I don't have a strong opinion here, but I like the fact that VS shows me the comment in a tooltip when I hover my mouse over these fields. When I am not familiar with the given source code, it makes it simpler for me.